assets/src/components/mapPage/mapDisplay.tsx

```diff
@@ -0,0 +1,391 @@
+      ? null
+      : routePattern?.shape
+
+  useEffect(() => {
+    setStateClasses(FollowerStatusClasses(followerState.shouldFollow))
+  }, [followerState.shouldFollow, setStateClasses])
+
+  return (
+    <>
+      {selectedVehicleOrGhost && (
+        <>
+          {showSelectionCard && (
+            <SelectionCardContainer>
+              <VehiclePropertiesCard
+                vehicleOrGhost={selectedVehicleOrGhost}
+                onClose={deleteSelection}
+              />
+            </SelectionCardContainer>
+          )}
+          {routePatternShape && <RouteShape shape={routePatternShape} />}
+          {isVehicle(selectedVehicleOrGhost) &&
+            (selectedVehicleOrGhost?.isShuttle ? (
+              <VehicleMarker
+                key={selectedVehicleOrGhost.id}
+                vehicle={selectedVehicleOrGhost}
+                isPrimary={true}
+                isSelected={true}
+              />
+            ) : (
+              <RouteVehicles
+                selectedVehicleRoute={selectedVehicleOrGhost.routeId}
+                selectedVehicleId={selectedVehicleOrGhost.id}
+                onPrimaryVehicleSelect={selectVehicle}
+              />
+            ))}
+        </>
+      )}
+
+      <InterruptibleFollower
+        onUpdate={onFollowerUpdate}
+        positions={position}
+        {...followerState}
+      />
+    </>
+  )
+}
+
+const SelectionDataLayers = ({
+  liveSelectedEntity,
+  showSelectionCard,
+  deleteSelection,
+  selectedRoutePattern,
+  setSelection,
+  setStateClasses,
+}: {
+  liveSelectedEntity: LiveSelectedEntity | null
+  showSelectionCard: boolean
+  selectedRoutePattern: RoutePattern | null
+  deleteSelection: () => void
+  setSelection: (selectedEntity: SelectedEntity | null) => void
+  setStateClasses: (classes: string | undefined) => void
+}) => {
+  switch (liveSelectedEntity?.type) {
+    case SelectedEntityType.Vehicle:
+      return (
+        <SelectedVehicleDataLayers
+          vehicleOrGhost={liveSelectedEntity.vehicleOrGhost}
+          showSelectionCard={showSelectionCard}
+          routePattern={selectedRoutePattern}
+          deleteSelection={deleteSelection}
+          selectVehicle={(vehicleOrGhost: VehicleOrGhost) =>
+            setSelection({
+              type: SelectedEntityType.Vehicle,
+              vehicleId: vehicleOrGhost.id,
+            })
+          }
+          setStateClasses={setStateClasses}
+        />
+      )
+    // TODO: handle SelectedEntityType.ROUTE
+    default:
+      return <MapElementsNoSelection setStateClasses={setStateClasses} />
+  }
+}
+
+const useSelectedEntityRoutePatterns = (
+  liveSelectedEntity: LiveSelectedEntity | null
+): [RoutePattern[] | null, RoutePattern | null] => {
+  const selectedEntityRouteData = selectedEntityRoutePattern(liveSelectedEntity)
+  const routePatterns: RoutePattern[] | null = useRoutePatterns(
+    selectedEntityRouteData?.routeId || null
+  )
```

Comment:
**suggestion:** if I understand correctly, regardless of whether we're dealing with a vehicle or route we will have an actual route pattern ID in `selectedEntityRouteData`, right? Setting aside shuttles, I suppose, but those don't have a route ID either. If that's the case, could we rework this so that instead there's a `useRoutePattern` hook that just takes the route pattern ID directly and queries the backend with that ID rather than route ID? That way we wouldn't have to `.find(...)` the right route pattern below based on its ID.

Reply:
Yeah that would be doable, and probably clearer! I was working with maybe a false assumption that fetching each route pattern shape could lead to some challenges keeping the selected state of the route pattern properties card in sync with the shape that is showing on the map. Though there is likely a more straight forward way to address that problem - I'll do some thinking and maybe set up some sync time after lunch!

Reply:
Based on our sync discussion, updated this to use a map of route patterns instead of a list.